src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -266,18 +303,35 @@ public final void visitToken(DetailAST blockCommentNode) {
                 result = TREE_CACHE.get().get(treeCacheKey);
             }
             else {
-                result = context.get().parser.parseJavadocAsDetailNode(blockCommentNode);
+                final JavadocDetailNodeParser.ErrorListenerOption errorListenerOption;
+                if (violateExecutionOnNonTightHtml || !doesCheckAcceptJavadocWithNonTightHtml()) {
+                    errorListenerOption = JavadocDetailNodeParser
+                            .ErrorListenerOption.NOTIFY_FOR_NON_TIGHT_HTML;
+                }
+                else {
+                    errorListenerOption = JavadocDetailNodeParser.ErrorListenerOption.DEFAULT;
+                }
+                result = context.get().parser
+                        .parseJavadocAsDetailNode(blockCommentNode, errorListenerOption);
                 TREE_CACHE.get().put(treeCacheKey, result);
             }
 
             if (result.getParseErrorMessage() == null) {
                 processTree(result.getTree());
             }
             else {
-                final ParseErrorMessage parseErrorMessage = result.getParseErrorMessage();
-                log(parseErrorMessage.getLineNumber(),
-                        parseErrorMessage.getMessageKey(),
-                        parseErrorMessage.getMessageArguments());
+                final DetailNode javadocTree = result.getTree();
+
+                if (javadocTree != null && doesCheckAcceptJavadocWithNonTightHtml()) {
```

Comment:
@Vladlis 
>yes, it is a typo, there should be `!result.isTightHtml()`

Yes but condition `!result.isTightHtml() && doesCheckAcceptJavadocWithNonTightHtml()` seems inappropriate since it requires javadoc to be non-tight and `doesCheckAcceptJavadocWithNonTightHtml()` to return `true`. But if `doesCheckAcceptJavadocWithNonTightHtml()` returns true then there is no point checking if javadoc is non-tight or tight since check is supposed to process all kinds of javadoc.

>Just add fields to ParseStatus instead of using error listener

But we can deduce everything we want from the variables we have at hand. So why add anything else ? 
Also, you mean to add `result.isTightHtml` to `ParseStatus` right ? Because to me, `ErrorListenerOption` should stay with `errorListener` as it looks to be a better design.